docs/source/conf.py

```diff
@@ -74,3 +105,145 @@
+# (source start file, target name, title,
+#  author, documentclass [howto, manual, or own class]).
+latex_documents = [
+    (
+        master_doc,
+        "rj_robocup.tex",
+        "rj\\_robocup Documentation",
+        "Oswin So",
+        "manual",
+    ),
+]
+
+# -- Options for manual page output ------------------------------------------
+
+# One entry per manual page. List of tuples
+# (source start file, name, description, authors, manual section).
+man_pages = [(master_doc, "rj_robocup", "rj_robocup Documentation", [author], 1)]
+
+# -- Options for Texinfo output ----------------------------------------------
+
+# Grouping the document tree into Texinfo files. List of tuples
+# (source start file, target name, title, author,
+#  dir menu entry, description, category)
+texinfo_documents = [
+    (
+        master_doc,
+        "rj_robocup",
+        "rj_robocup Documentation",
+        author,
+        "rj_robocup",
+        "One line description of project.",
+        "Miscellaneous",
+    ),
+]
+
+# -- Options for Epub output -------------------------------------------------
+
+# Bibliographic Dublin Core info.
+epub_title = project
+
+# The unique identifier of the text. This can be a ISBN number
+# or the project homepage.
+#
+# epub_identifier = ''
+
+# A unique identification for the text.
+#
+# epub_uid = ''
+
+# A list of files that should not be packed into the epub file.
+epub_exclude_files = ["search.html"]
+
+
+# -- Configuring standalone build ---------------------------------------------
+def configure_doxyfile(
+    input_dir: pathlib.Path,
+    output_dir: pathlib.Path,
+    doxyfile_dir: pathlib.Path,
+) -> None:
+    """Mimics the configure_file functionality in cmake.
+    :param input_dir: Input directory for Doxygen.
+    :param output_dir: Output directory for Doxygen.
+    :param doxyfile_dir: The directory of the Doxyfile.
+    """
+    print("doxyfile_dir: {}".format(doxyfile_dir))
+    try:
+        with open(doxyfile_dir / "Doxyfile.in", "r") as file:
+            filedata: str = file.read()
+    except Exception as e:
+        print(e)
+        raise e
+
+    print(1)
+    filedata = filedata.replace("@DOXYGEN_INPUT_DIR@", str(input_dir))
+    print(1)
+    filedata = filedata.replace("@DOXYGEN_OUTPUT_DIR@", str(output_dir))
+
+    print(3)
+    with open(doxyfile_dir / "Doxyfile", "w") as file:
+        file.write(filedata)
+
+
+def run_doxygen(doxyfile_dir: pathlib.Path) -> None:
+    """Runs the doxygen command in the passed in directory.
+    :param doxyfile_dir: Path to the directory containing the Doxyfile.
+    """
+    subprocess.call("doxygen", shell=True, cwd=doxyfile_dir)
+
+
+def generate_doxygen_xml(app: sphinx.application.Sphinx) -> None:
+    """Generates the doxygen XML for breathe.
+    :param app: Application object representing the Sphinx process
+    """
+    # Define all the paths.
+    cwd = pathlib.Path().resolve()
+    project_dir = cwd.parent.parent
```

Comment:
Actually I figured it out, since this was moved up a level this line must be changed. 
```suggestion
    project_dir = cwd.parent
```

Reply:
definitely test this more to see if it always works, but this fixed it for me.

Reply:
we probably should also add make clean-docs to the makefile

Reply:
^ same as make docs except `make clean` instead of `make html`

Reply:
can you run make html && make open-docs in the root `robocup-software/` directory instead? I wrote both with that as the intended spot to start from

Reply:
when I do that I get this error: 
```

[autosummary] generating autosummary for: autoapi/ball_classifier/index.rst, autoapi/index.rst, autoapi/rj_gameplay/basic_play_selector/index.rst, autoapi/rj_gameplay/eval/index.rst, autoapi/rj_gameplay/gameplay_node/index.rst, autoapi/rj_gameplay/index.rst, autoapi/rj_gameplay/play/basic_defense/index.rst, autoapi/rj_gameplay/play/basic_offense/index.rst, autoapi/rj_gameplay/play/defend_restart/index.rst, autoapi/rj_gameplay/play/defensive_clear/index.rst, ..., autoapi/test_skill/index.rst, autoapi/test_tactic/index.rst, autoapi/test_typed_key_dict/index.rst, autoapi/wall_calculations/index.rst, cmake.rst, contributing.rst, getting_started.rst, index.rst, meta_docs.rst, our_stack.rst
doxyfile_dir: /home/kasohrab/robocup/docs
[Errno 2] No such file or directory: '/home/kasohrab/robocup/docs/Doxyfile.in'

Extension error:
Handler <function generate_doxygen_xml at 0x7f5048667700> for event 'builder-inited' threw an exception (exception: [Errno 2] No such file or directory: '/home/kasohrab/robocup/docs/Doxyfile.in')
make[1]: *** [Makefile:20: html] Error 2
make[1]: Leaving directory '/home/kasohrab/robocup/robocup-software/docs'
make: *** [makefile:134: docs] Error 2
```
Changing the parent line fixes this for me. This makes sense to me as a solution since you moved conf.py up a directory level.

Reply:
also makefile always tells me `make: 'docs' is up to date.` Unless I run `make -B docs` (unconditionally make). So we should change the name of `make docs` to something else so the makefile doesn't always assume it is built. Or change the name of the docs folder if that is better

Reply:
Once I changed these two things though, it works great!

Reply:
Got it, committed your change and will figure out a better makefile name next week (I can't work on this without my ubuntu setup), maybe `make open-docs` should just build and open?

Reply:
note to self: should also make sure the `docs/build/` folder is added to .gitignore so people are never accidentally committing 2000+ build files to the repo (see mistake 2 commits down)

Reply:
so actually this change breaks RTD, see bottom of log: https://readthedocs.org/api/v2/build/16876134.txt

the `project_dir` should actually be ../.. since this conf.py file is in `robocup-software/docs/source/`, see [line 36 above](https://github.com/RoboJackets/robocup-software/pull/1897/files/499f02e040a162884dc86beb42e3204b68c99514#diff-008dcb3426febd767787b1521f1fe33086313b927ea37eaab86df5fa88a51698R36), so I'm not sure why changing this fixed local building

either way I will revert this for now until we can investigate the local errors further, since the RTD-hosted docs matter more here